hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/PrometheusMetricsSinkUtil.java

```diff
@@ -0,0 +1,200 @@
+   * @param unmodifiableTags list of metric tags
+   * @return modifiable list of metric tags
+   */
+  public static List<MetricsTag> addTags(String key, String username,
+      Collection<MetricsTag> unmodifiableTags) {
+    List<MetricsTag> metricTags = new ArrayList<>(unmodifiableTags);
+
+    addUsernameTag(username, metricTags);
+    addServerNameTag(key, metricTags);
+
+    return metricTags;
+  }
+
+  /**
+   * Adds a <tt>servername</tt> metric tag.
+   *
+   * @param key        metrics entry key
+   * @param metricTags list of metric tags
+   */
+  private static void addServerNameTag(String key,
+      List<MetricsTag> metricTags) {
+    if (!key.contains(UGI_METRICS)) {
+      return;
+    }
+
+    final String name = "servername";
+    final String description = "name of the server";
+    addTag(name, description, SERVER_NAME, metricTags);
+  }
+
+  /**
+   * Adds a <tt>username</tt> metric tag.
+   *
+   * @param username   caller username
+   * @param metricTags list of metric tags
+   */
+  private static void addUsernameTag(String username,
+      List<MetricsTag> metricTags) {
+
+    if (Strings.isNullOrEmpty(username)) {
+      return;
+    }
+
+    final String name = "username";
+    final String description = "caller username";
+    addTag(name, description, username, metricTags);
+  }
+
+  /**
+   * Adds a tag.
+   *
+   * @param name        tag name
+   * @param description tag description
+   * @param value       tag value
+   * @param metricsTags list of metric tags
+   */
+  private static void addTag(String name, String description, String value,
+      List<MetricsTag> metricsTags) {
+    MetricsInfo metricsInfo = createMetricsInfo(name, description);
+    MetricsTag metricTag = new MetricsTag(metricsInfo, value);
+    metricsTags.add(metricTag);
+  }
+
+  /**
+   * Creates MetricsInfo.
+   *
+   * @param name        metric name
+   * @param description metric description
+   * @return <tt>MetricsInfo</tt>
+   */
+  private static MetricsInfo createMetricsInfo(String name,
+      String description) {
+    return new MetricsInfo() {
+      @Override
+      public String name() {
+        return name;
+      }
+
+      @Override
+      public String description() {
+        return description;
+      }
+    };
+  }
+
+  /**
+   * Replaces default servername tag value with actual server name.
+   *
+   * @param metricEntryKey metric entry key
+   * @param metricKey      metric key
+   * @param serverPort     server port
+   * @return <tt>metricKey</tt> with replaced servername tag value.
+   */
+  public static String replaceServerNameTagValue(String metricEntryKey,
+      String metricKey, int serverPort) {
+    if (!metricEntryKey.contains(UGI_METRICS)) {
+      return metricKey;
+    }
+
+    String serverName = null;
+    switch (serverPort) {
+    case OMConfigKeys.OZONE_OM_HTTP_BIND_PORT_DEFAULT:
+      serverName = "OM";
+      break;
+    case ScmConfigKeys.OZONE_SCM_HTTP_BIND_PORT_DEFAULT:
```

Comment:
As @adoroszlai pointed out, this should be `ScmConfigKeys.OZONE_SCM_HTTP_ADDRESS_KEY` and from there get the port, in order to cover the case of a custom port.